docs/contributing.rst

```diff
@@ -1,3 +1,162 @@
+  - `tests/costs <../tests/costs>`_ tests for the solving costs.
+  - `tests/data <../tests/data>`_: tests for the simulated data module.
+  - `tests/datasets <../tests/datasets>`_: tests for the datasets.
+  - `tests/plotting <../tests/plotting>`_ tests for the plotting module.
+  - `tests/problems <../tests/problems>`_ tests for the problem classes, divided by problem type.
+  - `tests/solvers <../tests/solvers>`_ tests for the solvers.
+  - `tests/utils <../tests/utils>`_ tests for the utility functions.
+  - `tests/conftest.py <../tests/conftest.py>`_: ``pytest`` fixtures and utility functions.
+
+Code style guide
+----------------
+We rely on ``black`` and ``isort`` to do the most of the formatting - both of them are integrated as pre-commit hooks.
+You can use ``tox`` to check the changes::
+
+    tox -e lint
+
+Furthermore, we also require that:
+
+- functions are fully type-annotated.
+- exception messages are capitalized and end with ``.``.
+- warning messages are capitalized and do not end with ``.``.
+- when referring to variable inside an error/warning message, enclose its name in \`.
+- when referring to variable inside a docstrings, enclose its name in \``.
+
+Testing
+-------
+We use ``tox`` to automate our testing, as well as linting and documentation creation. To run the tests, run::
+
+    tox -e py{38,39,310}-{linux,macos}
+
+depending on the Python version(s) in your ``PATH`` and your operating system. We use ``flake8`` and ``mypy`` to further
+analyze the code. Use ``# noqa: <error1>,<error2>`` to ignore certain ``flake8`` errors and
+``# type: ignore[error1,error2]`` to ignore specific ``mypy`` errors.
+
+To run only a subset of tests, run::
+
+    tox -e <environment> -- <name>
+
+where ``<name>`` can be a path to a test file/directory or a name of a test function/class.
+For example, to run only the tests in the ``plotting`` module, use::
+
+    tox -e py39-linux -- tests/plotting/test_plotting.py
+
+If needed, a specific ``tox`` environment can be recreated as::
+
+    tox -e <environment> --recreate
+
+Writing documentation
+---------------------
+We use ``numpy``-style docstrings for the documentation with the following additions and modifications:
+
+- no type hints in the docstring (applies also for the return statement) are allowed,
+  since all functions are required to have the type hints in their signatures.
+- when referring to some argument within the same docstring, enclose that reference in \`\`.
+- prefer putting references in the ``references.bib`` instead under the ``References`` sections of the docstring.
+- use ``docrep`` for repeating documentation.
+
+In order to build the documentation, run::
+
+    tox -e docs
+
+Since the tutorials are hosted on a separate repository (see `Writing tutorials/examples`_), we download the newest
+tutorials/examples from there and build the documentation here.
+
+To validate the links inside the documentation, run::
+
+    tox -e check-docs
```

Comment:
we have `lint-docs` and `lint-code`  and same below we have `clean-docs` and `clean-code`

Reply:
no clean-code, just clean-docs